### DISCUSSION TITLE
Is there a way to put a notification ID or known when an specific notification was dismissed?

### DISCUSSION THREAD

**Discussion by distante** [Blackymas/NSPanel_HA_Blueprint#2333] [Q&A]
I have some persistent notification that I show in mobile phones, that have an "Proceed" button. I want to reproduce it on the NSPanel but I do not find any way to set an NotificationID that is not visible to the end user.

Is there some way to do this?

Thanks!

**Comment by edwardtfn** [comment]
Unfortunately this is not possible with the current implementation.

Could you please provide more details on how you wanna use this info, so I can think in a different implementation?

**Reply by distante** [reply]
Hi @edwardtfn, thank for your reply!

 My specific use case is as follows:

My kid has a condition that requires taking a medicine at night, so, at around 7pm, a Node Red recursive flow starts and sends persistent notifications to my phone, my wife's and my kid's room tablet about this and resend it each 5 minutes. I also have a local input_boolean that goes to false.

The notification has an "I took it" button, when someone tap it on any device, or, the notification is dismissed from all phones and the Node Red flow is stopped. 
I can react to it because on an Android notification I can set an action to that button.

Since in the NSPanel we do not have Ids, It is hard to show a "Take Pill" notification and react to it, because in the time between each remember (5 minute) a  completely different Notification could be shown. So just listening to the label or Text dismiss is not effective.

I have found a workaround though. I am manually storing "the last showed nspanel label" and on each update comparing it, if the last one was `label-pill` and the current is empty, I assume someone checked the Pill notification. But I am not sure if this can get some race condition.

---

I think what could be the best solution is to create a new event, something like `nspanel.accepted_notification` where we can send the requested id, or at least the label/text that was on the notification before being cleared.
Of course, this should not be called when we manually call `clear_notification`.

**Comment by dhottinger** [comment]
There is `binary_sensor.*_notification_unread`. It's disabled by default and needs to be enabled in the entity settings.
